### PR TITLE
Forbid prefix too

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

**Comment by greg0ire** [comment]
See https://github.com/doctrine/coding-standard/pull/31#discussion_r171371131

File: README.md [hunk 1]
```diff
@@ -19,8 +19,8 @@ exceptions/differences/extensions (:white_check_mark: are the implemented sniffs
 - Keep the nesting of control structures per method as small as possible
 - Prefer early exit over nesting conditions or using else
 - Abstract exception class names and exception interface names should be suffixed with `Exception`
-- :white_check_mark: Abstract classes should not be prefixed with `Abstract`
-- :white_check_mark: Interfaces should not be suffixed with `Interface`
+- :white_check_mark: Abstract classes should not be suffixed with `Abstract`
```
Comment (greg0ire) [inline README.md]: In fact, I'm not sure… do we, or do we not want to forbid the `Abstract` prefix. My PR was originally just about `Exception`, see with @carusogabriel why these other rules were added.

Comment (kukulich) [reply README.md]: The sniff reports both so this chane is not enough

Comment (greg0ire) [reply README.md]: @kukulich I don't know if we want to forbid the prefix or not.

Comment (Majkl578) [reply README.md]: It is currently forbidden in 4.0.0...

Comment (greg0ire) [reply README.md]: Oh well…

**Comment by ostrolucky** [comment]
please rebase

File: docs/en/reference/index.rst [hunk 2]
```diff
@@ -8,8 +8,8 @@ exceptions/differences/extensions.
 
 - Keep the nesting of control structures per method as small as possible
 - Abstract exception class names and exception interface names should be suffixed with ``Exception``
-- Abstract classes should not be prefixed with ``Abstract``
-- Interfaces should not be suffixed with ``Interface``
+- Abstract classes should not be suffixed or prefixed with ``Abstract``
+- Interfaces should not be suffixed or prefixed with ``Interface``
 - Concrete exception class names should not be suffixed with ``Exception``
```
Comment (gmponos) [inline docs/en/reference/index.rst]: Why not add prefix to exception as well?

**Comment by lcobucci** [comment]
We still need a sniff and tests here, right?

**Comment by ostrolucky** [comment]
This is documentation fix, not addition

**Comment by malarzm** [comment]
Wouldn't it be better to allow `Abstract` prefix instead? Or if not, how we're going to name abstract classes designed for inheritance?

**Comment by nikolaposa** [comment]
@malarzm At first glance this rule seems like a very hard to swallow, not only because makes it difficult to think of a name, but also the fact that one can forbid use of "Abstract" prefix but someone will always bypass that rule by using alternative one such as "Base", where this rule kinda loses the point. 

On second thoughts, the intent of this rule is to motivate similar strategy as in case of Interface suffix rule, which is tendency of using descriptive, clear names even for those base classes. I'll use Prooph Event Store components as an example of how I would name things:
- `EventStore` - an interface
- `PdoEventStore` - abstract class with logic common for PDO-based event store implementations
- `MySqlEventStore` - concrete implementation

IMO "Abstract" in `abstract class AbstractPdoEventStore` is excessive as "Interface" in `interface EventStoreInterface` definition.

File: docs/en/reference/index.rst (hunk 2 above)
Comment (Majkl578) [reply docs/en/reference/index.rst]: Note that no Exception prefix is currently not enforced by any current sniff.

**Pull request merged**